lib/entry.js

```diff
@@ -24,38 +26,27 @@ Entry.getOrCreate = function (id) {
   return entryMap[id] = entryMap[id] || new Entry(id);
 };
 
-var keySalt = 0;
-function makeUniqueKey() {
-  return Math.random()
-    .toString(36)
-    // Add an incrementing salt to help track key ordering and also
-    // absolutely guarantee we never return the same key twice.
-    .replace("0.", ++keySalt + ":");
-}
-
 Ep.addSetters = function (parent, setters, key) {
-  if (typeof key === "undefined") {
-    // If no key was provided, make a new unique key that won't collide
-    // with any other keys.
-    key = makeUniqueKey();
-  } else {
-    // If a key was provided, make sure it is distinct from keys provided
-    // by other parent modules.
-    key = parent.id + ":" + key;
-  }
-
   var names = Object.keys(setters);
   var nameCount = names.length;
 
+  if (! nameCount) {
+    return;
+  }
+
+  // If no key is provided, make a unique key. Otherwise, make sure the key is
+  // distinct from keys provided by other parent modules.
+  key = typeof key === "undefined"
+    ? makeUniqueKey()
+    : parent.id + ":" + key;
+
   for (var i = 0; i < nameCount; ++i) {
     var name = names[i];
     var setter = setters[name];
-    if (typeof setter === "function" &&
-        // Ignore any requests for the exports.__esModule property."
-        name !== "__esModule") {
```

Comment:
Since both string and symbols are non-enumerable they'll never show up in an `Object.keys` result so the checks aren't needed.